starter-repos/shared-ci-cd/appctl/validate.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+validate-manifests:
+  stage: validate-manifests
+  image: gcr.io/cloud-solutions-images/appctl
+  script:
+  - |
+    cat > /root/.appctlconfig <<EOF
+    personal_access_tokens:
+    - host: ${CI_SERVER_HOST}
+      token: ${GIT_ACCESS_TOKEN}
+    EOF
+  - cat > /root/.ssh/id_rsa <<EOF
+    ${MANIFEST_WRITER_KEY}
+    EOF
+  - chmod 400 /root/.ssh/id_rsa
+  - echo "StrictHostKeyChecking no " > /root/.ssh/config
+  - git config --global user.email ${CI_PROJECT_NAME}-ci@${CI_SERVER_HOST}
+  - git config --global user.name ${CI_PROJECT_NAMESPACE}/${CI_PROJECT_NAME}
+  - |
+    appctl init ${APP_NAME} --app-config-repo ${APP_REPO_URL} \
+    --deployment-repo ${DEPLOYMENT_REPO_URL}
```

Comment:
Maybe define the line 20-33 as a new stage `init-application` and call it in .gitlab-ci.yml or in both `validate-manifest` and `prepare-manifest`

Reply:
SG. Will update it.

Reply:
`validate-manifests` or `prepare-manifests` needs to share the same docker container configured in the `init-application` stage.
To achieve this, we have to have a stage to build the docker image, save it to a shared cache and then reuse the container and then load the container.
It may cause issues if the container is too big.
See discussions here: https://forum.gitlab.com/t/passing-docker-image-between-build-and-test-stage-in-gitlab-runner/2444/7.

I would prefer to duplicate the configuration.

Reply:
OK